### PR TITLE
feat: add HCP callout to sidecar

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

For products with an HCP offering, adds a `Try HCP` callout to the sidecar of integration pages.

## 🛠️ How

- Implements a new `TryHcpCalloutSidecarPlacement` component
    - This component accepts a `productSlug`, and handles the logic of whether to render the HCP callout, and what content to show
- Adds `TryHcpCalloutSidecarPlacement` to integrations views
    - Adds to `ProductIntegrationLayout`, which covers the majority of integration pages
    - ~Adds to~ Does not add to `ProductIntegrationsLanding`, as this is not in designs

## 📸 Design Screenshots

| View | Before | After |
| - | - | - | 
| [ReadmeView][/waypoint/integrations/aws-ami] | readme-before | readme-after |
| [ComponentView][/waypoint/integrations/aws-ami/latest/components/builder] | component-before | component-after |

## 🧪 Testing

> **Note**: The logic to display the HCP callout only for products that have HCP offerings has been tested in #1376. Vault and Waypoint are the only products that currently have integrations pages, so are the only integrations sections we can test at the moment. 

- [ ] Visit each of the views listed in the section above, for [Vault][/vault/integrations] and [Waypoint][/waypoint/integrations]
    - Each view should show a Try HCP callout in the sidecar
    - The product theme of the Try HCP callout should match as expected
    - The Try HCP callout should link to the expected location

## 💭 Anything else?

Not at the moment!

[preview]: https://dev-portal-git-zsintegrations-hcp-callout-in-sidecar-hashicorp.vercel.app/
[task]: https://app.asana.com/0/1202097197789424/1203709011206839/f
[/waypoint/integrations]: https://dev-portal-git-zsintegrations-hcp-callout-in-sidecar-hashicorp.vercel.app/waypoint/integrations
[/vault/integrations]: https://dev-portal-git-zsintegrations-hcp-callout-in-sidecar-hashicorp.vercel.app/vault/integrations
[/waypoint/integrations/aws-ami/latest/components/builder]: https://dev-portal-git-zsintegrations-hcp-callout-in-sidecar-hashicorp.vercel.app/waypoint/integrations/aws-ami/latest/components/builder
[/waypoint/integrations/aws-ami]: https://dev-portal-git-zsintegrations-hcp-callout-in-sidecar-hashicorp.vercel.app/waypoint/integrations/aws-ami

### CODE REVIEW

File: src/pages/[productSlug]/integrations/index.tsx [hunk 1]
```diff
@@ -48,7 +52,9 @@ export function generateProductIntegrationLibrarySidebarNavData(
 	}
 }
 
-export async function getServerSideProps({ params }) {
+export async function getServerSideProps({
+	params,
+}): Promise<GetStaticPropsResult<ViewProps & { metadata: HeadMetadataProps }>> {
```
Comment (zchsh) [inline src/pages/[productSlug]/integrations/index.tsx:57]: One tangential change in this PR was adding a return type here to the props. Felt like the right thing to do given I was adding in a `productSlug` property to the returned props 👍

**Comment by zchsh** [comment]
@BrandonRomano Smol-ish PR to add the Try HCP callout to the sidecar on all integrations pages. Let me know what you think 🙇

File: src/components/try-hcp-callout/components/try-hcp-callout-sidecar-placement/index.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { TryHcpCalloutCompact } from '..'
+import { tryHcpCalloutContent, hasHcpCalloutContent } from '../../content'
+
+/**
+ * A wrapper around TryHcpCalloutCompact.
+ *
+ * Handles conditionally rendering HCP callouts only for products with an
+ * HCP offering. Also populates the callout with pre-built content.
+ */
+export function TryHcpCalloutSidecarPlacement({
```
Comment (BrandonRomano) [inline src/components/try-hcp-callout/components/try-hcp-callout-sidecar-placement/index.tsx:10]: This is nice 😎 👏

File: src/views/product-integrations-landing/index.tsx [hunk 3]
```diff
@@ -4,26 +4,31 @@ import { IntegrationsSearchProvider } from './contexts/integrations-search-conte
...
+	productSlug,
 }: ViewProps) {
 	return (
 		<IntegrationsSearchProvider integrations={integrations}>
 			<SidebarSidecarLayout
 				sidebarNavDataLevels={sidebarNavDataLevels}
 				breadcrumbLinks={breadcrumbLinks}
-				sidecarSlot={<></>}
+				sidecarSlot={
+					<TryHcpCalloutSidecarPlacement productSlug={productSlug} />
```
Comment (BrandonRomano) [inline src/views/product-integrations-landing/index.tsx:30]: I don't think we want this on the landing page, actually!

It looks like we actually shouldn't even have a sidecar slot for anything on that landing page now that I am re-reviewing the designs, but don't worry about that for now (I'll make another ticket and chat with design to figure out what they're looking for there).

**Review by BrandonRomano — APPROVED** [review]
Small change, just remove from the landing page.  Looks good other than that, so feel free to remove & merge.

File: src/views/product-integrations-landing/index.tsx (hunk 3 above)
Comment (zchsh) [reply src/views/product-integrations-landing/index.tsx]: Ah great catch - i've removed the callout from `/<product>/integrations` landing pages ✅ 🔪  will keep an eye on the board for that new ticket ☺️

**Pull request merged**